full/src/main/java/apoc/load/LoadHtml.java

```diff
@@ -34,48 +37,61 @@
-
-                output.put(key, getElements(elements, config));
+            query.keySet().forEach(key -> {
+                        Elements elements = document.select(query.get(key));
+                        output.put(key, getElements(elements, config));
             });
 
-            return Stream.of( new MapResult(output) );
-        } catch(Exception e){
-            throw new RuntimeException("Can't read the HTML from: "+ url);
+            return Stream.of(new MapResult(output) );
+        } catch (FileNotFoundException e) {
+            throw new RuntimeException("File not found from: " + url);
+        } catch(UnsupportedEncodingException e) {
+            throw new RuntimeException("Unsupported charset: " + charset);
+        } catch(Exception e) {
+            throw new RuntimeException("Can't read the HTML from: "+ url, e);
         }
     }
 
     private List<Map<String, Object>> getElements(Elements elements, Map<String, Object> config) {
+        boolean failSilently = Util.toBoolean(config.getOrDefault("failSilently", false));
         List<Map<String, Object>> elementList = new ArrayList<>();
 
         for (Element element : elements) {
-            Map<String, Object> result = new HashMap<>();
-            if(element.attributes().size() > 0) result.put("attributes", getAttributes(element));
-            if(!element.data().isEmpty()) result.put("data", element.data());
-            if(!element.val().isEmpty()) result.put("value", element.val());
-            if(!element.tagName().isEmpty()) result.put("tagName", element.tagName());
-
-            if ( Util.toBoolean( config.getOrDefault("children", false) ) ) {
-                if(element.hasText())  result.put("text", element.ownText());
-
-                result.put("children", getElements(element.children(), config));
-            }
-            else {
-                if(element.hasText()) result.put("text", element.text());
+            try {
+                    Map<String, Object> result = new HashMap<>();
+                    if(element.attributes().size() > 0) result.put("attributes", getAttributes(element));
+                    if(!element.data().isEmpty()) result.put("data", element.data());
+                    if(!element.val().isEmpty()) result.put("value", element.val());
+                    if(!element.tagName().isEmpty()) result.put("tagName", element.tagName());
+
+                    if (Util.toBoolean(config.getOrDefault("children", false))) {
+                        if(element.hasText()) result.put("text", element.ownText());
+
+                        result.put("children", getElements(element.children(), config));
+                    }
+                    else {
+                        if(element.hasText()) result.put("text", element.text());
+                    }
+
+                    elementList.add(result);
+            } catch (Exception e) {
+                final String parseError = "Error during parsing element: " + element;
```

Comment:
maybe errors could be added to the output, in a reserved key? (maybe not for the scope of this PR/issue though)

Reply:
So, you mean, a structure like {`a: [listOfTagA]`, `h1: [list]`, .... `errors: [listOfErrorsIfExist]`} , right?
In this case maybe the `log.warn` at line 90 is useless

Reply:
Perhaps, we could leave both the log and the list and transform `failSilently` from boolean to enum, for example `FALSE, WITH_LOG, WITH_LIST`, and with `WITH_LOG`, we remains the `log.warn`, and with `WITH_LIST` we create the `errors: [listOfErrorsIfExist]`

Reply:
sounds good!

Reply:
@fbiville Changed as previously described